platform/ios/test/MGLSourceTests.m

```diff
@@ -0,0 +1,24 @@
+#import "MGLMapViewTests.h"
+
+@interface MGLSourceTests : MGLMapViewTests
+
+@end
+
+@implementation MGLSourceTests
+
+- (void)testDuplicateSources {
+    MGLVectorSource *source1 = [[MGLVectorSource alloc] initWithIdentifier:@"my-source" URL:[NSURL URLWithString:@"mapbox://mapbox.mapbox-terrain-v2"]];
+    MGLVectorSource *source2 = [[MGLVectorSource alloc] initWithIdentifier:@"my-source" URL:[NSURL URLWithString:@"mapbox://mapbox.mapbox-terrain-v2"]];
+    
+    [self.mapView.style addSource: source1];
+    
+    @try {
```

Comment:
I think that once this change is merged into the iOS release branch it might make more sense to do these tests in `MGLStyleTests` along with what has been added in https://github.com/mapbox/mapbox-gl-native/pull/7272/files#diff-ce3b7fe93109fd552f9627d0dcc0ceb4R107 for checking for duplicate instances.